createsend/createsend.py

```diff
@@ -144,12 +142,24 @@ def make_request(self, method, path, params={}, body="", username=None,
       # Check that the actual url which would be requested matches self.faker.url.
       actual_url = "https://%s%s" % (parsed_base_uri.netloc, self.build_url(parsed_base_uri, path, params))
       self.faker.actual_url = actual_url
-      if self.faker.url != actual_url:
+      def same_urls(url_a, url_b):
+          a = urlparse(url_a)
+          b = urlparse(url_b)
+          return (a.scheme == b.scheme and
+                  a.netloc == b.netloc and
+                  a.path == b.path and
+                  a.params == b.params and
+                  parse_qs(a.query) == parse_qs(b.query) and
+                  a.fragment == b.fragment
+                  )
+      if not same_urls(self.faker.url, actual_url):
         raise Exception("Faker's expected URL (%s) doesn't match actual URL (%s)" % (self.faker.url, actual_url))
 
       self.faker.actual_body = body
+      def same_bodies(body_a, body_b):
+          return json.loads(body_a) == json.loads(body_b)
       if self.faker.body is not None:
-        if self.faker.body != body:
```

Comment:
Comparing json results also because of hash randomization being enabled by default.